File: src/javascript/example/cli/src/requests/do-cool-stuff.js
import sessionless from 'sessionless-node';
import superagent from 'superagent';
import config from '../../config/local.js';

const doCoolStuff = async (bodyFromRegistration) => {
  const { uuid, welcomeText, color } = bodyFromRegistration;

  const colorURL = config.colors[color].serverURL;
  const colorSignaturePlacement = config.colors[color].signature;

  let message = {
    uuid,
    coolness: 'max',
    timestamp: new Date().getTime() + ''
  };

// console.log(message);

  let signature = await sessionless.sign(JSON.stringify(message));

  signature = signature.length % 2 === 1 ? '0' + signature : signature;

// console.log(signature.length);
// console.log(signature);

  let post = superagent.post(colorURL + '/cool-stuff');

  if(colorSignaturePlacement === 'payload') {
    const payload = {
      ...message,
      signature
    };
    post = post.send(payload)
  } else {
    post = post.send(message)
               .set('signature', signature);
  }
  return post.set('Content-Type', 'application/json')
             .set('Accept', 'application/json')
         .then(res => {
           res.body.color = color;
           return res.body;
         })
         .catch(err => console.log(`Error: ${err}`));
};

export default doCoolStuff;

File: src/python/examples/cli/src/requests/do_cool_stuff.py
from sessionless import SessionlessSecp256k1
import requests

async def do_cool_stuff(registration_tuple):
    print('registered!')
    print('Gonna just do blue for now')

    json = registration_tuple[0].json()

    print(f'json looks like: {json}')
    print(f'uuid: {json['uuid']}')

    uuid = json['uuid']
    private_key = registration_tuple[1]
    public_key = registration_tuple[2]
    
    sessionless = SessionlessSecp256k1()

    timestamp = 'right now' #we'll deal with time later

    def get_key():
        return private_key

    message_to_sign = f'{{"uuid":"{uuid}","coolness":"max","timestamp":"{timestamp}"}}'

    signature = await sessionless.sign(message_to_sign.encode('ascii'), get_key)

    message = {
        "uuid": uuid,
        "coolness": "max",
        "timestamp": timestamp,
        "signature": signature
    }

    headers = {'content-type': 'application/json'}
    response = requests.post('http://127.0.0.1:3001/cool-stuff', headers=headers, json=message) # update this to be dynamic
    
    print(f'response: {response}')

    return response
